File: services/api-service/app/models/system_config.py
from datetime import datetime
from typing import Any, Optional
from uuid import UUID

from pydantic import BaseModel


class SystemConfigBase(BaseModel):
    category: str
    key: str
    value: Any
    description: Optional[str] = None
    is_active: bool = True


class SystemConfigCreate(SystemConfigBase):
    pass


class SystemConfigUpdate(BaseModel):
    value: Optional[Any] = None
    description: Optional[str] = None
    is_active: Optional[bool] = None


class SystemConfig(SystemConfigBase):
    id: UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class SystemConfigResponse(BaseModel):
    config: SystemConfig
    message: str = "Configuration retrieved successfully"

File: frontend/src/types/system-config.ts
export interface SystemConfig {
  id: string
  category: string
  key: string
  value: any
  description?: string
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface SystemConfigCreate {
  category: string
  key: string
  value: any
  description?: string
  is_active?: boolean
}

export interface SystemConfigUpdate {
  value?: any
  description?: string
  is_active?: boolean
}
